zerver/views/users.py

```diff
@@ -493,6 +492,9 @@ def create_user_backend(
     password: str=REQ(),
     full_name_raw: str=REQ("full_name"),
 ) -> HttpResponse:
+    if not user_profile.is_api_super_user:
+        return json_error(_("User not authorized for this query"))
```

Comment:
I think we want `@require_realm_admin` too, don't we?  Otherwise demoting someone from realm administrator access would not remove this permission.

Also, we need to update the API docs for this endpoint.

Reply:
Oh, I see, the macros covers the API docs issue.  OK.

Reply:
Hmm makes sense, but on the other hand - if we don't want a demoted is_api_super_user admin to have this power, then we probably don't want them to have the message forging power either. Two approaches I can see:
1. Make  `is_api_super_user` into a `@property`  - only be True if the appropriate flag in the databse is `True` and `role` is `ROLE_REALM_ADMINISTRATOR` or higher
2. Require realm admin on this endpoint and limit message forging to cross realm bots only (because has anyone used message forging using this API on Cloud? self-hosted deployments can use cross-realm bots to forge, so this shouldn't be too limiting)

Reply:
Well, ultimately we're likely to change this into multiple flags.  Maybe we should just do that now?  
* `can_forge_sender` for the message-send feature
* `can_create_users` for this privilege

I think `can_create_users` should require admin status, but the other might be best to not require it (no reason to give unnecessary permissions there).